src/nvidia.cc

```diff
@@ -309,10 +334,35 @@ struct nvidia_s {
 	ATTR_ID attribute;
 	char *token;
 	SEARCH_ID search;
+//  added new field for GPU id
+    int gpu_id;
+};
+
+//Cache by value
+struct nvidia_c_value {
+    int memtotal = -1;
+	int gputempthreshold = -1;
+};
```

Comment:
I am struggling with a lot of topic and one is what framework to use. Those changes were done using gedit out of the box.

Maybe you can give me some hints about what framework to use for developing? 

Actually I am using Ubuntu 16.04LT as it is certified by NVIDIA CUDA Framework. It was from the certified distributions, the one with higher linux kernel version.

I want the framework to solve my little issues like having multiple NVidia. Might be I get some Radeon Vega too in an heterogeneous environment if this the case I will add a module for them. 

About my main project I am not sure if I will branch GRUB2 or KERNEL, maybe both it is a boolean OR.

As I said those are the requirements I have in mind, if you have any idea about what framework to use it will be really welcome.